yarn-project/noir-contracts/src/contracts/schnorr_auth_witness_account_contract/src/main.nr

```diff
@@ -9,117 +19,58 @@ contract SchnorrAuthWitnessAccount {
+        // let actions = AccountActions::private(&mut context, ACCOUNT_ACTIONS_STORAGE_SLOT, is_valid_impl);
+        actions.entrypoint(payload);
     }
 
     #[aztec(private)]
-    fn is_valid(
-        message_hash: Field
-    ) -> Field {
-        if (_inner_is_valid(message_hash, context.this_address())){
-            0xe86ab4ff
-        } else {
-            0
-        }
+    fn is_valid(message_hash: Field) -> Field {
+        let actions = AccountActions::private(&mut context, ACCOUNT_ACTIONS_STORAGE_SLOT, is_valid_impl);
+        actions.is_valid(message_hash)
     }
 
     #[aztec(public)]
-    fn is_valid_public(
-        message_hash: Field,
-    ) -> Field {
-        let storage = Storage::init(Context::public(&mut context));
-        let value = storage.approved_action.at(message_hash).read();
-        if (value == 1){
-            0xe86ab4ff
-        } else {
-            0
-        }
+    fn is_valid_public(message_hash: Field) -> Field {
+        let actions = AccountActions::public(&mut context, ACCOUNT_ACTIONS_STORAGE_SLOT, is_valid_impl);
+        actions.is_valid_public(message_hash)
     }
 
     #[aztec(private)]
-    fn set_is_valid_storage(
-        message_hash: Field,
-        value: Field,
-    ) {
-        assert((value == 0) | (value == 1), "value must be a boolean");
-        assert(context.msg_sender() == context.this_address(), "only the owner can set the storage");
-        // assert(_inner_is_valid(message_hash, context.this_address()), "only the owner can set the storage");
-
-        let selector = compute_selector("_set_is_valid_storage(Field,Field)");
-        let _void = context.call_public_function(context.this_address(), selector, [message_hash, value]);
+    fn set_is_valid_storage(message_hash: Field, value: Field) {
+        let actions = AccountActions::private(&mut context, ACCOUNT_ACTIONS_STORAGE_SLOT, is_valid_impl);
+        actions.set_is_valid_storage(message_hash, value)
     }
 
     #[aztec(public)]
-    internal fn _set_is_valid_storage(
-        message_hash: Field,
-        value: Field,
-    ) {
-        let storage = Storage::init(Context::public(&mut context));
-        storage.approved_action.at(message_hash).write(value);
+    internal fn internal_set_is_valid_storage(message_hash: Field, value: Field) {
+        let actions = AccountActions::public(&mut context, ACCOUNT_ACTIONS_STORAGE_SLOT, is_valid_impl);
+        actions.internal_set_is_valid_storage(message_hash, value)
     }
 
-    fn _inner_is_valid(
-        message_hash: Field,
-        address: Field,
-    ) -> pub bool{
+    fn is_valid_impl(message_hash: Field, address: Field) -> pub bool{
         let witness = get_auth_witness(message_hash);
         assert(recover_address(message_hash, witness) == address);
         true
     }
+
+    fn get_actions(context: Context) -> pub AccountActions {
+        AccountActions::init(context, ACCOUNT_ACTIONS_STORAGE_SLOT, is_valid_impl)
+    }
```

Comment:
@kevaundray this is the function that was causing the error `Params to the program should only contains numbers and arrays`